src/Compiler/Compiler.h

```diff
@@ -245,6 +245,9 @@ class Compiler {
   void ClbPackingOption(ClbPacking clbPacking);
   ClbPacking ClbPackingOption() const;
 
+  void SetCompilerName(const std::string& name) { m_compilerName = name; }
```

Comment:
this method should not exists. What if I create CompilerFPGA and call it 'dummy' ? Everything will be broken.
Set name in the compiler constructor.

Reply:
So why it is here? Maybe you want to finalize it first?

Reply:
It is draft PR now. Sure, let me finalize them and let you know.